pkg/scalers/aws/aws_config_cache.go

```diff
@@ -0,0 +1,131 @@
+package aws
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/credentials/stscreds"
+	"github.com/aws/aws-sdk-go-v2/service/sts"
+	"github.com/go-logr/logr"
+	"golang.org/x/crypto/sha3"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
```

Comment:
Could you please add here general comment what is this file about? Just a few sentences